tests/plugins/autocomplete/autocomplete.js

```diff
@@ -97,6 +97,99 @@
+			editable.fire( 'keydown', new CKEDITOR.dom.event( { keyCode: 9 } ) ); // TAB
+
+			assert.areEqual( '<p>item1</p>', editor.getData() );
+
+			ac.destroy();
+		},
+
+		'test custom single key inserts match': function() {
+			var editor = this.editor, bot = this.editorBot, editable = editor.editable(),
+				ac = new CKEDITOR.plugins.autocomplete( this.editor, matchTestCallback, dataCallback );
+
+			bot.setHtmlWithSelection( '' );
+
+			ac.commitKeystroke = 16; // SHIFT
+
+			editor.editable().fire( 'keyup', new CKEDITOR.dom.event( {} ) );
+
+			editable.fire( 'keydown', new CKEDITOR.dom.event( { keyCode: 40 } ) ); // ARROW DOWN
+			editable.fire( 'keydown', new CKEDITOR.dom.event( { keyCode: 16 } ) ); // SHIFT
+
+			assert.areEqual( '<p>item1</p>', editor.getData() );
+
+			ac.destroy();
+		},
+
+		'test custom keys inserts match': function() {
+			var editor = this.editor, bot = this.editorBot, editable = editor.editable(),
+				ac = new CKEDITOR.plugins.autocomplete( this.editor, matchTestCallback, dataCallback );
+
+			bot.setHtmlWithSelection( '' );
+
+			ac.commitKeystroke.push( 16 ); // SHIFT
+
+			editor.editable().fire( 'keyup', new CKEDITOR.dom.event( {} ) );
+
+			editable.fire( 'keydown', new CKEDITOR.dom.event( { keyCode: 40 } ) ); // ARROW DOWN
+			editable.fire( 'keydown', new CKEDITOR.dom.event( { keyCode: 16 } ) ); // SHIFT
+
+			assert.areEqual( '<p>item1</p>', editor.getData() );
+
+			ac.destroy();
+		},
+
+		'test custom keys inserts match (global configuration)': function() {
+			var editor = this.editor, bot = this.editorBot, editable = editor.editable(),
+				configKeyCodes = CKEDITOR.config.autocomplete_commitKeystroke.slice();
+
+			bot.setHtmlWithSelection( '' );
+
+			CKEDITOR.config.autocomplete_commitKeystroke.push( 16 ); // SHIFT
+
+			var ac = new CKEDITOR.plugins.autocomplete( this.editor, matchTestCallback, dataCallback );
+
+			editor.editable().fire( 'keyup', new CKEDITOR.dom.event( {} ) );
+
+			editable.fire( 'keydown', new CKEDITOR.dom.event( { keyCode: 40 } ) ); // ARROW DOWN
+			editable.fire( 'keydown', new CKEDITOR.dom.event( { keyCode: 16 } ) ); // SHIFT
+
+			assert.areEqual( '<p>item1</p>', editor.getData() );
+
+			ac.destroy();
+
+			CKEDITOR.config.autocomplete_commitKeystroke = configKeyCodes;
+		},
+
+		'test custom single key inserts match (global configuration)': function() {
+			var editor = this.editor, bot = this.editorBot, editable = editor.editable();
+
+			bot.setHtmlWithSelection( '' );
+
+			CKEDITOR.config.autocomplete_commitKeystroke = 16; // SHIFT
+
+			var ac = new CKEDITOR.plugins.autocomplete( this.editor, matchTestCallback, dataCallback );
+
+			editor.editable().fire( 'keyup', new CKEDITOR.dom.event( {} ) );
+
+			editable.fire( 'keydown', new CKEDITOR.dom.event( { keyCode: 40 } ) ); // ARROW DOWN
+			editable.fire( 'keydown', new CKEDITOR.dom.event( { keyCode: 16 } ) ); // SHIFT
+
+			assert.areEqual( '<p>item1</p>', editor.getData() );
+
+			ac.destroy();
+		},
+
```

Comment:
Lastly I don't see a case that checks that if `autocomplete_commitKeystroke` was changed to something like `config.autocomplete_commitKeystroke = [ x, y ]` (here both are other than `9` which is tab key) then tab key no longer causes the insertion.